core/sql/cli/Cli.cpp

```diff
@@ -9517,20 +9517,37 @@ Lng32 SQLCLI_LOBddlInterface
   switch (qType)
     {
     case LOB_CLI_CREATE:
+    case LOB_CLI_ALTER :
       {
-	// create lob metadata table
-	str_sprintf(query, "create ghost table %s (lobnum smallint not null, storagetype smallint not null, location varchar(4096) not null, column_name varchar(256 bytes) character set utf8, primary key (lobnum)) ",lobMDName);
-	 lobDebugInfo(query,0,__LINE__,lobTrace);
+       
+        // create lob metadata table
+        str_sprintf(query, "create ghost table %s (lobnum smallint not null, storagetype smallint not null, location varchar(4096) not null, column_name varchar(256 bytes) character set utf8, primary key (lobnum)) ",lobMDName);
+        lobDebugInfo(query,0,__LINE__,lobTrace);
 
-	// set parserflags to allow ghost table
-	currContext.setSqlParserFlags(0x1);
-	
-	cliRC = cliInterface->executeImmediate(query);	
-	currContext.resetSqlParserFlags(0x1);
+        // set parserflags to allow ghost table
+        currContext.setSqlParserFlags(0x1);
 	
-	if (cliRC < 0)
-	    goto error_return;
+        cliRC = cliInterface->executeImmediate(query);	
+        currContext.resetSqlParserFlags(0x1);
 	
+        if (cliRC < 0)
+          {
+            ComDiagsArea *tempDiags = ComDiagsArea::allocate(currContext.exHeap());
```

Comment:
Do you have to deallocate tempDiags?

Reply:
decrRefCount will deallocate the diags if refcount goes down to 0.  So inn this case we are good- it will get dealllocated since refCount is equal to 1.